src/AdoPat/PatManager.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// Manage the interaction between the PAT cache and Azure DevOps.
+    /// </summary>
+    public class PatManager
+    {
+        private const int ValidToExtensionDays = 7;
+        private const int NearingExpirationDays = 2;
+
+        private IPatCache cache;
+        private IPatClient client;
+        private Func<DateTime> now;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PatManager"/> class.
+        /// </summary>
+        /// <param name="cache">Any class that implements the <see cref="IPatCache"/> interface.</param>
+        /// <param name="client">Any class that implements the <see cref="IPatClient"/> interface.</param>
+        /// <param name="now">A function for computing the current moment. Defaults to null, which
+        /// uses <see cref="DateTime.UtcNow"/>. Overriding this should only be necessary in testing.</param>
+        public PatManager(IPatCache cache, IPatClient client, Func<DateTime> now = null)
+        {
+            this.cache = cache;
+            this.client = client;
+            this.now = now ?? new Func<DateTime>(() => DateTime.UtcNow);
+        }
+
+        /// <summary>
+        /// Fetches a PAT matching the given options from the cache, creating or regenerating a new PAT as necessary.
+        /// </summary>
+        /// <param name="options">Options used to match the PAT in the cache or when creating/regenerating a new one.</param>
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        /// <returns>An Azure DevOps Personal Access Token.</returns>
+        public async Task<PatToken> GetPatAsync(
+            PatOptions options,
+            CancellationToken cancellationToken = default)
+        {
+            bool writeBack = false;
+            var pat = this.cache.GetPat(options.ToString());
+
+            if (pat == null || await this.Inactive(pat, cancellationToken).ConfigureAwait(false))
+            {
+                var patTokencreateRequest = new PatTokenCreateRequest
+                {
+                    DisplayName = options.DisplayName,
+                    Scope = string.Join(" ", options.Scopes),
+                    ValidTo = this.now().AddDays(ValidToExtensionDays),
+                    AllOrgs = default,

```

Comment:
I think explicitly saying false, even though yes that's the default, is more clear here. 
```suggestion
                    AllOrgs = false,

```

Reply:
It is unclear from reading this code that it's a bool, if you use default.

Reply:
(granted Intellisense would fix this in VS to some degree).

Reply:
I thought about this and used `default` here because we're getting guidance from Azure DevOps folks to use `false` right now. By using `default` we will always use whatever they say is the default.

However, I don't suspect they'll ever change their minds about this and I'm happy to just use `false`.

Reply:
Instead, I think I'm going to make it so that `PatClient.CreateAsync` only takes primitives and internally generates a `PatTokenCreateRequest` with `AllOrgs` set to `false` using the existing `PatClient.AllOrgs` constant.

Reply:
For posterity - `default` here actually does _not_ use the `[DefaultValue(false)]` attribute defined on that class for that field, it's taking the default value for the `bool` type which happen to be the same here. The attribute declares the default value to use when instantiating an instance of that struct/class and the value is not given. So - we are going to pass `false` ourselves, since we know we never want to request an all orgs pat.